internal/workload/podman/podman.go

```diff
@@ -55,17 +74,46 @@ func (p *Podman) Remove(workloadId string) error {
 	return nil
 }
 
-func (p *Podman) Run(manifestPath, authFilePath string) ([]string, error) {
+// getContainerDetails returns some basic information about a container
+func (p *Podman) getContainerDetails(containerId string) (*ContainerReport, error) {
+	data, err := containers.Inspect(p.podmanConnection, containerId, nil)
+	if err != nil {
+		return nil, err
+	}
+	// this is the default one
+	network, ok := data.NetworkSettings.Networks["podman"]
+	if !ok {
+		return nil, fmt.Errorf("Cannot retrieve container '%s' network information", containerId)
+	}
+
+	return &ContainerReport{
+		IPAddress: network.InspectBasicNetworkConfig.IPAddress,
+		Id:        containerId,
+		Name:      data.Name,
+	}, nil
+}
+
+func (p *Podman) Run(manifestPath, authFilePath string) ([]*PodReport, error) {
 	options := play.KubeOptions{
 		Authfile: &authFilePath,
 	}
 	report, err := play.Kube(p.podmanConnection, manifestPath, &options)
 	if err != nil {
 		return nil, err
 	}
-	var podIds []string
-	for _, pod := range report.Pods {
-		podIds = append(podIds, pod.ID)
+
+	var podIds = make([]*PodReport, len(report.Pods))
+	for i, pod := range report.Pods {
+		report := &PodReport{Id: pod.ID}
+		for _, container := range pod.Containers {
+			c, err := p.getContainerDetails(container)
```

Comment:
Yes, there is no way to get the list of all containers with IPS in a single call :-(